fourth_task.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+/**
+ * Function to read input data
+ */
+function readHttpLikeInput() {
+    $f = fopen( 'php://stdin', 'r' );
+    $store = "";
+    $toread = 0;
+
+    while( $line = fgets( $f ) ) {
+        $store .= preg_replace("/\r/", "", $line);
+
+        if (preg_match('/Content-Length: (\d+)/',$line,$m)) 
+            $toread=$m[1]*1; 
+
+        if ($line == "\r\n") 
+              break;
+    }
+
+    if ($toread > 0) 
+        $store .= fread($f, $toread);
+
+    return $store;
+}
+
+$contents = readHttpLikeInput();
+
+/**
+ * Function simplifies results printing by formatting it to needed look
+ * 
+ * @param string $statuscode status of response
+ * @param string $statusMessage status message of response
+ * @param array $headers headers of response
+ * @param string $body body of response
+ */
+function outputHttpResponse($statuscode, $statusmessage, $headers, $body) {
+    
+    echo "HTTP/1.1 $statuscode $statusmessage" . PHP_EOL;
+
+    foreach($headers as $key => $value) {
+        echo "$key: $value" . PHP_EOL;
+    }
+
+    echo PHP_EOL . $body;
+}
+
+/**
+ * Function works with http request to define
+ * what response should be given
+ * 
+ * @param string $method method of request
+ * @param string $uri uri of request
+ * @param array $headers headers of request
+ * @param string $body body of request
+ */
+function processHttpRequest($method, $uri, $headers, $body) {
+
+    // Define needed headers
+    $generatedHeaders = [
+        "Server" => "Apache/2.2.14 (Win32)",
+        "Content-Length" => "",
+        "Connection" => "Closed",
+        "Content-Type" => "text/html; charset=utf-8"
+    ];
+
+    // If uri or header incorrect - return client error
+    if ($uri !== "/api/checkLoginAndPassword" || $headers["Content-Type"] !== "application/x-www-form-urlencoded") {
+        $statuscode = "400";
```

Comment:
for such things as response code, message etc - you should use constants, or enums.